internal/dao/rbac_policy.go

```diff
@@ -159,18 +160,30 @@ func (p *Policy) fetchRoleBindingSubjects(kind, name string) ([]string, error) {
 	if err != nil {
 		return nil, err
 	}
+
+	ns, n := client.Namespaced(name)
 	ss := make([]string, 0, len(rbs))
 	for _, rb := range rbs {
 		for _, s := range rb.Subjects {
-			if s.Kind == kind && s.Name == name {
-				ss = append(ss, rb.RoleRef.Kind+":"+rb.Name)
+			if isSameSubject(kind, ns, n, s) {
+				ss = append(ss, rb.RoleRef.Kind+":"+rb.RoleRef.Name)
 			}
 		}
 	}
 
 	return ss, nil
 }
 
+// isSameSubject verifies if the incoming type name and namespace match a subject from a
+// cluster/roleBinding. A ServiceAccount will always have a namespace and needs to be validated to ensure
+// we don't display permissions for a ServiceAccount with the same name in a different namespace
+func isSameSubject(kind, namespace, name string, subject rbacv1.Subject) bool {
+	if kind == "ServiceAccount" {
+		return subject.Kind == kind && subject.Name == name && subject.Namespace == namespace
```

Comment:
Probably should short circuit here ie if kind and name don't match return false right off the bat

Reply:
Swapped the check around